daffodil-io/src/main/scala/org/apache/daffodil/io/InputSource.scala

```diff
@@ -357,18 +372,25 @@ class BucketingInputSource(inputStream: java.io.InputStream, bucketSize: Int = 1
 
   def position(bytePos0b: Long): Unit = {
     val (bucketIndex, _) = bytePositionToIndicies(bytePos0b)
-    Assert.invariant(bucketIndex >= oldestBucketIndex && bucketIndex < buckets.length)
+    Assert.invariant(bucketIndex < buckets.length)
     curBytePosition0b = bytePos0b
   }
 
   def lockPosition(bytePos0b: Long): Unit = {
     val (bucketIndex, _) = bytePositionToIndicies(bytePos0b)
-    Assert.invariant(bucketIndex >= oldestBucketIndex && bucketIndex < buckets.length)
+    Assert.invariant(bucketIndex < buckets.length)
+    if (buckets(bucketIndex.toInt) == null)
+      throw new BacktrackingException
+
     buckets(bucketIndex.toInt).refCount += 1
   }
 
   def releasePosition(bytePos0b: Long): Unit = {
     val (bucketIndex, _) = bytePositionToIndicies(bytePos0b)
+
+    if (buckets(bucketIndex.toInt) == null)
+      return
+
```

Comment:
Instead of returning, does it make sense to ignore the refCount for that bucket, but then continue to release buckets? I guess there now might be a bunch of releaseble buckets, but I guess also who cares? Since the next read will result in the backtrack exception? If it doesn't make sense to just say who care and not try to release, this could maybe use a comment as to why we don't care.